src/server/server_state.h

```diff
@@ -130,6 +130,8 @@ class ServerState {  // public struct - to allow initialization.
     gstate_ = s;
   }
 
+  uint64_t GetCachedUsedMemory(uint64_t now_ns);
```

Comment:
Cache can mean different things. As a start, I'd add a comment describing what this method does. Reading this first, I thought that this might be the memory of the items in cache (as in cache mode).
Perhaps call it `GetLastUsedMemory()`? I'd even call it `GetUsedMemory()` and comment that it might be stale for some short period of time..